tests/Tokenizer/TokensAnalyzerTest.php

```diff
@@ -670,4 +670,49 @@ class TestClass {
 
         return $cases;
     }
+
+    /**
+     * @dataProvider provideIsWhilePartOfDoWhile
+     */
+    public function testIsWhilePartOfDoWhile($source, $index, $expected)
```

Comment:
TODO missing test for exception when index passed for token not of kind `T_WHILE`